pkg/liveshare/port_forwarder.go

```diff
@@ -99,6 +111,43 @@ func (fwd *PortForwarder) Forward(ctx context.Context, conn io.ReadWriteCloser)
 	return awaitError(ctx, errc)
 }
 
+// Connects to and pings a given address to ensure that the server is shared and the port is forwarded.
+func WaitForPortConnection(ctx context.Context, address string) error {
+	waitCtx, cancel := context.WithTimeout(ctx, connectionTimeout)
+	g, waitCtx := errgroup.WithContext(waitCtx)
+	defer cancel()
+
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+
+	g.Go(func() error {
+		for {
+			select {
+			case <-waitCtx.Done():
+				return fmt.Errorf("timed out waiting for connection")
+			case <-ticker.C:
+				// Verify that the port can be connected to
+				conn, err := net.Dial("tcp", address)
+				if err != nil {
+					continue
+				}
+
+				defer conn.Close()
+
+				// Send a ping and make sure it succeed
+				_, err = conn.Write([]byte("ping"))
+				if err != nil {
+					continue
+				}
+
+				return nil
+			}
```

Comment:
Instead of ticking every 1 sec, we can do it faster doing this:

```
select {
case <-ctx.Done():
	...
default:
	..
}
```

This will constantly ping the server, until we timeout without any waiting in between

Reply:
Wasn't sure if there were any downsides to constantly spamming the port, updated 🙂